### PR TITLE
Clean up some code

### PR DESCRIPTION
**Pull request by iology**
This may prevent some memory corruption. Also added more checks for NULL pointers.

### CODE REVIEW

**Comment by iology** [comment]
I feel bad for the unchecked memory allocation everywhere, just hope the NULL is never spawned. :(

File: Main/Source/nonhuman.cpp [hunk 1]
```diff
@@ -960,9 +960,10 @@ truth vampirebat::SpecialBiteEffect(character* Victim, v2 HitPos, int BodyPartIn
 }
 
 bool ChamaleonPolymorphRandomly(chameleon* c){
-  if(!c->StateIsActivated(POLYMORPH_LOCK)){
```
Comment (fejoa) [inline Main/Source/nonhuman.cpp]: Did you accidentally nerf this feature (I think it allows a chameleon to wear a ring of unchanging(?) and remain in that form). Or is this inherently a bug-fix?

Comment (iology) [reply Main/Source/nonhuman.cpp]: That's already checked in `character::PolymorphRandomly()` https://github.com/Attnam/ivan/blob/1cb0d19f79117832ca1d099af28785238216f3f5/Main/Source/char.cpp#L6601-L6610 unless we want to optimize it a bit or be more explicit.

Comment (fejoa) [reply Main/Source/nonhuman.cpp]: Ahh I see, I should've seen that earlier. Great!

**Comment by fejoa** [comment]
I decided to resolve the conflict due to #543 using github's resolve feature. @iology can you please check it is resolved correctly?

**Comment by iology** [comment]
No, git showed a confusing diff:

```diff
--- a/Main/Source/nonhuman.cpp
+++ b/Main/Source/nonhuman.cpp
@@ -959,7 +959,7 @@ truth vampirebat::SpecialBiteEffect(character* Victim, v2 HitPos, int BodyPartIn
     return false;
 }

-bool ChamaleonPolymorphRandomly(chameleon* c){
+bool ChameleonPolymorphRandomly(chameleon* c){
   if(!c->StateIsActivated(POLYMORPH_LOCK)){
     character* NewForm = c->PolymorphRandomly(100, 1000, 500 + RAND() % 500);
     if(NewForm==NULL)ABORT("chameleon PolymorphRandomly failed"); //means needs more checks
```

The actual function name is still Cham**a**leonPolymorphRandomly.

File: Main/Source/nonhuman.cpp [hunk 2]
```diff
@@ -959,10 +959,11 @@ truth vampirebat::SpecialBiteEffect(character* Victim, v2 HitPos, int BodyPartIn
...
 
-bool ChameleonPolymorphRandomly(chameleon* c){
-  if(!c->StateIsActivated(POLYMORPH_LOCK)){
-    character* NewForm = c->PolymorphRandomly(100, 1000, 500 + RAND() % 500);
-    if(NewForm==NULL)ABORT("chameleon PolymorphRandomly failed"); //means needs more checks
+bool ChamaleonPolymorphRandomly(chameleon* c){
+  character* NewForm = c->PolymorphRandomly(100, 1000, 500 + RAND() % 500);
+
+  if(NewForm != NULL)
+  {
```
Comment (iology) [inline Main/Source/nonhuman.cpp]: The conflict got resolved into this. 👅

Comment (fejoa) [reply Main/Source/nonhuman.cpp]: Oh no :sob: Sorry @iology, can you please fix? (This is really the last time I use that resolve conflicts feature, I think I'm going to write to GitHub).

Comment (iology) [reply Main/Source/nonhuman.cpp]: I rebased the branch.

**Pull request merged**